packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -44,37 +46,35 @@ function resolvePropertySignature(
   entry: ApiEntryPoint,
   p: ApiPropertySignature
 ) {
-  if (isRecordType(p)) {
+  const typeExcerpt = p.propertyTypeExcerpt.spannedTokens[0];
+
+  if (isRecordType(typeExcerpt)) {
     return buildEntityFromProperty(p);
   }
 
-  if (isPropertyUsingTypeAlias(p)) {
+  if (isTypeAlias(typeExcerpt)) {
     return buildEntityFromPropertyAndResolveTypeAlias(entry, p);
   }
 
-  if (isReference(p)) {
+  if (isReference(typeExcerpt)) {
     return buildObjEntityFromProperty(entry, p);
   }
 
   return buildEntityFromProperty(p);
 }
 
-function isPropertyUsingTypeAlias(m: ApiPropertySignature) {
-  const {canonicalReference} = m.propertyTypeExcerpt.spannedTokens[0];
-  const canonicalRef = canonicalReference?.toString() || '';
+function isTypeAlias(token: ExcerptToken) {
+  const canonicalRef = token.canonicalReference?.toString() || '';
   return /:type$/.test(canonicalRef);
```

Comment:
Would replacing this with `return token.canonicalReference.symbol?.meaning === 'type'` be equivalent?

Reply:
It would be even better, since `meaning` is a typed string that can only be one of a few values. Great suggestion 💯